src/test/spec/v2_runner/mod.rs

```diff
@@ -126,10 +133,22 @@ pub async fn run_v2_test(test_file: TestFile) {
             EventClient::with_additional_options(options, None, test.use_multiple_mongoses, None)
                 .await;
 
-        let _fp_guard = match test.fail_point {
-            Some(fail_point) => Some(fail_point.enable(client.deref(), None).await.unwrap()),
-            None => None,
-        };
+        // TODO RUST-900: Remove this extraneous call.
+        if test.operations.iter().any(|op| op.name == "distinct") {
```

Comment:
As a slight optimization, we can also add conditions here only to run `distinct` on each host when we're working with a 4.2 sharded cluster, as that's the only topology where we run into this issue.

Reply:
That sounds good. As an aside, some of the other drivers have also run into issues related to `StaleDbVersion` on other versions. However, it sounds like these bugs are weird and driver-specific.

Since *we* only ran into the issue on 4.2 sharded, I think it makes sense to add this check, but I want to draw attention to the fact that the issue has impacted other versions on other drivers.